tests/PHPUnit/Integration/CronArchive/QueueConsumerTest.php

```diff
@@ -1461,6 +1472,38 @@ public function getTestDataForShouldSkipArchiveBecauseLowerPeriodOrSegmentIsInPr
             'archiveToProcess' => ['name' => 'done5f4f9bafeda3443c3c2d4b2ef4dffadc', 'idsite' => 1, 'date1' => '2020-03-04', 'date2' => '2020-03-04', 'period' => Day::PERIOD_ID, 'segment' => 'browserCode==IE'],
             'expected' => null
         ];
+
+        yield 'day period should not be detected as intersecting when range is processed' => [
+            'existingInvalidations' => [
+                ['name' => 'done', 'idsite' => 5, 'date1' => '2020-03-03', 'date2' => '2020-03-05', 'period' => Range::PERIOD_ID, 'status' => 1, 'ts_started' => date('Y-m-d H:i:s')],
+            ],
+            'archiveToProcess' => ['name' => 'done', 'idsite' => 5, 'date1' => '2020-03-04', 'date2' => '2020-03-04', 'period' => Day::PERIOD_ID],
+            'expected' => null,
+        ];
+
+        yield 'range period should be detected as intersecting when day is processed' => [
+            'existingInvalidations' => [
+                ['name' => 'done', 'idsite' => 5, 'date1' => '2020-03-04', 'date2' => '2020-03-04', 'period' => Day::PERIOD_ID, 'status' => 1, 'ts_started' => date('Y-m-d H:i:s')],
+            ],
+            'archiveToProcess' => ['name' => 'done', 'idsite' => 5, 'date1' => '2020-03-03', 'date2' => '2020-03-05', 'period' => Range::PERIOD_ID],
+            'expected' => 'lower or same period in progress (period = day, date = 2020-03-04)',
+        ];
+
+        yield 'range period should be detected as intersecting when overlapping range is processed' => [
```

Comment:
I'm wondering if that behavior is actually needed. Processing ranges should not use other ranges, so there might be no need to wait for an intersecting range period though. Doesn't need to be changed in this PR, but we could maybe create a follow up issue if we want to change that.

Reply:
I would probably keep it, it's a good reminder that even ranges can overlap, either fully or partially, but yeah, they wouldn't use nested ranges (if they existed) to get the data. Hmmm... In a worst case scenario it can delay data if it's waiting for one range to finish before starting the other one if e.g. all days are already available.

Reply:
We should keep that test in any case. It was purely around challenging the fact that the current code behavior might delay data processing unnecessarily for such cases.